mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala

```diff
@@ -468,7 +469,16 @@ object LocalLDAModel extends Loader[LocalLDAModel] {
       val topics = Range(0, k).map { topicInd =>
         Data(Vectors.dense((topicsDenseMatrix(::, topicInd).toArray)), topicInd)
       }
-      spark.createDataFrame(topics).repartition(1).write.parquet(Loader.dataPath(path))
+
+      val bufferSize = Utils.byteStringAsBytes(
+        spark.conf.get("spark.kryoserializer.buffer.max", "64m"))
+      // We calculate the approximate size of the model
+      // We only calculate the array size, considering an
+      // average string size of 15 bytes, the formula is:
+      // (floatSize * vectorSize + 15) * numWords
+      val approxSize = (4L * k + 15) * topicsMatrix.numRows
+      val nPartitions = ((approxSize / bufferSize) + 1).toInt
+      spark.createDataFrame(topics).repartition(nPartitions).write.parquet(Loader.dataPath(path))
```

Comment:
why not? i think it does works. the multiple parquet files will may be in random order, but it will save topic indices. when u call load process, parquet will restore dataframe, u can check the LocalLDAModel's load method, it will scan all dataframe's row with the topic indices to rebuild the (topic x vocab) breeze matrix.